km61/Marchenko_Oleksandr/laboratory1.sql

```diff
@@ -1,2 +1,103 @@
--- LABORATORY WORK 1
--- BY Marchenko_Oleksandr
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+оновлювати дані в таблицях
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATE USER Marchenko IDENTIFIED BY Marchenko;
+
+TEMPORARY TEMPLATE "TEMPSPACE"; 
+DEFAULT USER "USERSPACE";
+
+GRANT "CONNECT" TO Marchenko;  
+GRANT UPDATE ANY TABLE TO Marchenko;
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Людина дивиться кіно.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE TABLE Films
+(
+    filmName VARCHAR(30)
+);
+ALTER TABLE Films
+(
+
+);
+
+CREATE TABLE Humanoid
+(
+    humanName VARCHAR(30)
+);
+
+ALTER TABLE Humanoid
+(
+
+);
+
+CREATE TABLE CinemaHistory
+(
+    
+);
+ALTER TABLE CinemaHistory
+(
+
+);
+ALTER TABLE CinemaHistory
+(
+
+);
+ALTER TABLE CinemaHistory
+(
+
+);
+
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+GRANT CREATE ANY TABLE TO Marchenko;
+GRANT SELECT ANY TABLE TO Marchenko;
+GRANT INSERT ANY TABLE TO Marchenko; --OR ADD
+CREATE DATABASE SomeName();
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Скільком покупцям продано найдешевший товар?
+Виконати завдання в SQL. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+/*---------------------------------------------------------------------------
+3.b. 
+Яка країна, у якій живуть постачальники має найдовшу назву?
+Виконати завдання в SQL. 
+4 бали
+
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+/*---------------------------------------------------------------------------
+c. 
+Вивести ім’я та країну покупця, як єдине поле client_name, для тих покупців, що мають не порожнє замовлення.
+Виконати завдання в алгебрі Кодда. 
+4 бали
+
```

Comment:
SELECT
    TRIM(cust_name) || ' from ' ||
    TRIM(cust_country) as client_name
FROM 
    Customers
WHERE
    Customers.cust_id IN (
        SELECT 
            DISTINCT(cust_id)
        FROM 
            Orders
        );